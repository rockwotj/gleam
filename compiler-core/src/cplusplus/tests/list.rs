use crate::assert_cpp;

#[test]
fn make_list() {
    assert_cpp!(
        r#"
pub fn palindrome(x: String, y: String, z: String) -> List(String) {
    [x, y, z, y, x]
}"#
    );
}
