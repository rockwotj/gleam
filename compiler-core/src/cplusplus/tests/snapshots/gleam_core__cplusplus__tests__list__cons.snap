---
source: compiler-core/src/cplusplus/tests/list.rs
expression: "\npub fn cons(head: String, tail: List(String)) -> List(String) {\n    [head, ..tail]\n}"
---
#ifndef MY_PACKAGE_MY_MODULE_H_
#define MY_PACKAGE_MY_MODULE_H_

#include <gleam.h>

namespace my {
namespace module {

gleam::Ref<gleam::List> cons(gleam::String head, gleam::Ref<gleam::List<gleam::String>> tail);

} // namespace my
} // namespace module

#endif // MY_PACKAGE_MY_MODULE_H_

---
#include "module.h"

namespace my {
namespace module {

gleam::Ref<gleam::List> cons(gleam::String head, gleam::Ref<gleam::List<gleam::String>> tail) {
  return gleam::MakeList<gleam::String>({head},
  tail);
};

} // namespace my
} // namespace module

